server/room-battle.ts

```diff
@@ -726,6 +728,14 @@ export class RoomBattle extends RoomGames.RoomGame {
 		for (const player of this.players) player.wantsTie = false;
 
 		switch (lines[0]) {
+		case 'requesteddata':
+			lines = lines.slice(1);
+			if (this.dataResolvers && this.dataResolvers.length > 0) {
+				const resolver = this.dataResolvers.shift();
+				if (resolver) resolver(lines);
+			}
```

Comment:
```suggestion
			const resolver = this.dataResolvers!.shift()!;
			resolver(lines);
```

I suspect that if `resolver` doesn't exist, that's a bug that should invoke the crashlogger.

Reply:
A minor issue is: what if the battle is destroyed while a resolver is waiting? Some Promises would hang forever.

This is extremely obscure: there are only two possibilities here: a battle is destroyed after 10 minutes of activity (unlikely for a resolver to wait that long), or a battle process hard-crashes (bypassing crashlogger; this can happen from OOMs, from externally killing them, or possibly from stack overflows).

Of course I'd accept this PR without the fix, but if you have time, you might want to support rejecting the Promise with "battle destroyed" in this case.

Reply:
Will do.